sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -36,22 +46,26 @@ def __init__(self, topic_hostname, credential, **kwargs):
         self._topic_hostname = topic_hostname
         auth_policy = _get_authentication_policy(credential)
         self._client = EventGridPublisherClientImpl(authentication_policy=auth_policy, **kwargs)
-
+  
     def send(self, events, **kwargs):
-        # type: (Union[List[CloudEvent], List[EventGridEvent], List[CustomEvent]], Any) -> None
+        # type: (SendType, Any) -> None
         """Sends event data to topic hostname specified during client initialization.
 
         :param events: A list of CloudEvent/EventGridEvent/CustomEvent to be sent.
         :type events: Union[List[models.CloudEvent], List[models.EventGridEvent], List[models.CustomEvent]]
         :rtype: None
          """
+        if not isinstance(events, list):
+            events = [events]
 
-        if all(isinstance(e, CloudEvent) for e in events):
+        if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(e) for e in events):
+            kwargs.setdefault("content_type", "application/cloudevents-batch+json; charset=utf-8")
             self._client.publish_cloud_event_events(self._topic_hostname, events, **kwargs)
-        elif all(isinstance(e, EventGridEvent) for e in events):
+        elif all(isinstance(e, EventGridEvent) for e in events) or all(isinstance(e, dict) for e in events):
```

Comment:
intentionally not doing a client side validation for dicts that are not cloud event